tests/layering/helpers_test.go

```diff
@@ -0,0 +1,611 @@
+	for _, file := range files {
+		out, err := ExecCmdOnNode(cs, node, []string{"stat", file})
+		require.Nil(t, err, "expected to find %s on node %s: %s", file, node.Name, out.String())
+		t.Logf("found %s on node %s (this was as expected)", file, node.Name)
+	}
+}
+
+// Asserts that a node does not have files in the expected places.
+func assertNodeNotHasFiles(t *testing.T, cs *framework.ClientSet, node *corev1.Node, files []string) {
+	for _, file := range files {
+		out, err := ExecCmdOnNode(cs, node, []string{"stat", file})
+		require.NotNil(t, err, "expected not to find %s on node %s:\n%s", file, node.Name, out.String())
+		t.Logf("did not find %s on node %s (this was as expected)", file, node.Name)
+	}
+}
+
+// Determines if a node is ready.
+func isNodeReady(node *corev1.Node) bool {
+	for _, condition := range node.Status.Conditions {
+		if condition.Type == corev1.NodeReady && condition.Status == "True" {
+			return true
+		}
+	}
+
+	return false
+}
+
+// RebootAndWait reboots a node, waits until it's status is ready.
+func rebootAndWait(t *testing.T, cs *framework.ClientSet, node *corev1.Node) error {
+	t.Logf("rebooting %s", node.Name)
+
+	// Get an updated node object since this one may potentially be out-of-date
+	updatedNode, err := cs.Nodes().Get(context.TODO(), node.ObjectMeta.Name, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("could not get updated node: %w", err)
+	}
+
+	prevBootID := updatedNode.Status.NodeInfo.BootID
+
+	if _, err := ExecCmdOnNode(cs, node, []string{"systemctl", "reboot"}); err != nil {
+		return fmt.Errorf("could not reboot node: %w", err)
+	}
+
+	startTime := time.Now()
+	err = wait.Poll(2*time.Second, 10*time.Minute, func() (bool, error) {
+		n, err := cs.Nodes().Get(context.TODO(), node.ObjectMeta.Name, metav1.GetOptions{})
+		require.Nil(t, err)
+
+		if n.Status.NodeInfo.BootID != prevBootID {
+			return isNodeReady(n), nil
+		}
+		return false, nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("node %q never rebooted (waited %s)", node.ObjectMeta.Name, time.Since(startTime))
+	}
+
+	t.Logf("node %q has rebooted (waited %s)", node.ObjectMeta.Name, time.Since(startTime))
+
+	return nil
+}
+
+// Ensures that rpm-ostree is running by first interrogating its status from
+// systemd, then tries to start it if it is not running. It will try to start
+// rpm-ostree up to five times.
+func ensureRPMOstreeIsRunning(t *testing.T, cs *framework.ClientSet, node *corev1.Node) error {
+	activeState := "ActiveState=active"
+	startLimitHit := "Result=start-limit-hit"
+
+	runner := NewNodeCmdRunner(cs, node, mcoNamespace)
+
+	runOpts := NodeCmdOpts{
+		Command: []string{"systemctl", "show", "--no-page", "rpm-ostreed"},
+		Retries: 5,
+		RetryCheckFunc: func(attempt int, cr *CmdResult, err error) bool {
+			// We ran into an error trying to get the rpm-ostreed status, retry.
+			if err != nil {
+				return false
+			}
+
+			combined := string(cr.Stdout) + string(cr.Stderr)
```

Comment:
do we need a space or colon or something here ie ...+ " " + string(...)

Reply:
I don't think we do, but it couldn't hurt!

Reply:
I have a slightly better idea to do that, somewhat inspired by how `os/exec` does it.